packages/core/src/agent/didcomm/versions/v2/sicpa-didcomm/DIDResolverService.ts

```diff
@@ -0,0 +1,142 @@
+import type { VerificationMethod, DidDocumentService } from '../../../../../modules/dids/domain'
+import type { AgentContext } from '../../../../context'
+import type { DIDDoc, DIDResolver } from 'didcomm'
+
+import { InjectionSymbols } from '../../../../../constants'
+import { AriesFrameworkError } from '../../../../../error'
+import { DidCommV1Service, DidCommV2Service, IndyAgentService } from '../../../../../modules/dids/domain'
+import { DidResolverService } from '../../../../../modules/dids/services/DidResolverService'
+import { inject, injectable } from '../../../../../plugins'
+import { AgentContextProvider } from '../../../../context'
+
+@injectable()
+export class DIDResolverService implements DIDResolver {
+  private agentContext: AgentContext | undefined
+  private agentContextProvider: AgentContextProvider
+  private resolverService: DidResolverService
+
+  public constructor(
+    @inject(InjectionSymbols.AgentContextProvider) agentContextProvider: AgentContextProvider,
+    resolverService: DidResolverService
+  ) {
+    this.agentContextProvider = agentContextProvider
+    this.resolverService = resolverService
+  }
+
+  // FIXME: Find better way for managing agent context
+  public bindAgentContext(agentContext: AgentContext): DIDResolverService {
+    this.agentContext = agentContext
+    return this
+  }
```

Comment:
As this class can't be stateless and we can't provide the agentContext in the specific methods we should make this a statefull class and create a new instance for each agent context that is not a singleton. We can register class as ContextScoped in the module where we register this service and in that case we can inject the AgentContext in the constrcutor